sstables/sstables.cc

```diff
@@ -3302,8 +3302,8 @@ class sstable::s3_storage final : public sstable::storage {
     future<> ensure_remote_prefix(const sstable& sst);
 
 public:
-    s3_storage(sstring host, s3::endpoint_config_ptr cfg, sstring bucket, sstring dir)
-        : _client(s3::client::make(ipv4_addr(host)))
+    s3_storage(sstring endpoint, s3::endpoint_config_ptr cfg, sstring bucket, sstring dir)
+        : _client(s3::client::make(std::move(endpoint)))
```

Comment:
maybe endpoint should be folded into endpoint_config_ptr?

Reply:
Yes, it's easy, but what's the point? It'll make less arguments to s3_storage() constructor at the cost of the { endpoint -> endpoint config } map storing two copies of each endpoint string.

Reply:
I don't see how endpoint is different from endpoint_config, that's all.

Reply:
And the map could become a set (though I don't think it's worth the effort).